Compilation breaks on kernel 6.x

I'm running Arch with the latest kernel (6.3.1) and trying to compile this module throws some errors.

```
❯ make
make -C /lib/modules/6.3.1-arch1-1/build M=/home/user/bin/slimbook-qc71-dkms/slimbook-qc71 modules
  CC [M]  /home/user/bin/slimbook-qc71-dkms/slimbook-qc71/ec.o
  CC [M]  /home/user/bin/slimbook-qc71-dkms/slimbook-qc71/features.o
  CC [M]  /home/user/bin/slimbook-qc71-dkms/slimbook-qc71/main.o
  CC [M]  /home/user/bin/slimbook-qc71-dkms/slimbook-qc71/misc.o
  CC [M]  /home/user/bin/slimbook-qc71-dkms/slimbook-qc71/pdev.o
  CC [M]  /home/user/bin/slimbook-qc71-dkms/slimbook-qc71/events.o
  CC [M]  /home/user/bin/slimbook-qc71-dkms/slimbook-qc71/debugfs.o
  CC [M]  /home/user/bin/slimbook-qc71-dkms/slimbook-qc71/battery.o
/home/user/bin/slimbook-qc71-dkms/slimbook-qc71/battery.c:93:27: error: initialization of ‘int (*)(struct power_supply *, struct acpi_battery_hook *)’ from incompatible pointer type ‘int (*)(struct power_supply *)’ [-Werror=incompatible-pointer-types]
   93 |         .add_battery    = qc71_laptop_batt_add,
      |                           ^~~~~~~~~~~~~~~~~~~~
/home/user/bin/slimbook-qc71-dkms/slimbook-qc71/battery.c:93:27: note: (near initialization for ‘qc71_laptop_batt_hook.add_battery’)
/home/user/bin/slimbook-qc71-dkms/slimbook-qc71/battery.c:94:27: error: initialization of ‘int (*)(struct power_supply *, struct acpi_battery_hook *)’ from incompatible pointer type ‘int (*)(struct power_supply *)’ [-Werror=incompatible-pointer-types]
   94 |         .remove_battery = qc71_laptop_batt_remove,
      |                           ^~~~~~~~~~~~~~~~~~~~~~~
/home/user/bin/slimbook-qc71-dkms/slimbook-qc71/battery.c:94:27: note: (near initialization for ‘qc71_laptop_batt_hook.remove_battery’)
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:252: /home/user/bin/slimbook-qc71-dkms/slimbook-qc71/battery.o] Error 1
make[1]: *** [Makefile:2025: /home/user/bin/slimbook-qc71-dkms/slimbook-qc71] Error 2
make: *** [Makefile:25: all] Error 2
```

Let me know if I can provide any more info.

--- a/slimbook-qc71/battery.c
+++ b/slimbook-qc71/battery.c
@@ -72,7 +72,7 @@ static struct attribute *qc71_laptop_batt_attrs[] = {
 };
 ATTRIBUTE_GROUPS(qc71_laptop_batt);
 
-static int qc71_laptop_batt_add(struct power_supply *battery)
+static int qc71_laptop_batt_add(struct power_supply *battery, struct acpi_battery_hook *hook)
 {
 	if (strcmp(battery->desc->name, "BAT0") != 0)
 		return 0;
@@ -80,7 +80,7 @@ static int qc71_laptop_batt_add(struct power_supply *battery)
 	return device_add_groups(&battery->dev, qc71_laptop_batt_groups);
 }
 
-static int qc71_laptop_batt_remove(struct power_supply *battery)
+static int qc71_laptop_batt_remove(struct power_supply *battery, struct acpi_battery_hook *hook)
 {
 	if (strcmp(battery->desc->name, "BAT0") != 0)
 		return 0;

